internal/host/host.go

```diff
@@ -327,7 +327,7 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
 		return common.NewApiError(http.StatusNotFound, err)
 	}
 
-	if m.Config.BootstrapHostMAC != "" && !h.Bootstrap {
+	if cluster != nil && m.Config.BootstrapHostMAC != "" && !h.Bootstrap {
```

Comment:
Maybe it would be cleaner to refactor this a bit.

We could move the infraEnv query above the `if h.ClusterID != nil` block.
Then put this entire block inside that one.

That way it's very clear that this shouldn't run if we can't find a clusterID.

Alternatively it feels like an error case to have `BootstrapHostMAC` set and not have a clusterID. Maybe we should check for that explicitly and just fail? @zaneb what do you think?

Reply:
I did unset the clusterid while running [this unit test](https://github.com/openshift/assisted-service/blob/e27247fe07ace5f6212ea5896871faa84e329fe7/internal/host/host_test.go#L1176), sorry I guess I didn't fully understand late-binding and the usage of this Bootstrap MAC. 

If this scenario can't happen then I'll go ahead and close this issue and PR.